### DISCUSSION TITLE
Why not implement InstantID for SDXL? Better results than the IpAdapter alone...

### DISCUSSION THREAD

**Discussion by motolo** [Acly/krita-ai-diffusion#1003] [Ideas]
After testing and testing, I believe that the implementation of InstantID, carried out by Matteo, is a better solution for faceswap and for the style it can produce, better results than the IPadapter alone.

 I think adding this node and a menu addition is not difficult.
 https://github.com/cubiq/ComfyUI_InstantID
 https://www.youtube.com/watch?v=wMLiGhogOPE

From the Wiki:
InstantID requires insightface, you need to add it to your libraries together with onnxruntime and onnxruntime-gpu.

The InsightFace model is antelopev2 (not the classic buffalo_l). Download the models (for example from [here](https://drive.google.com/file/d/18wEUfMNohBJ4K3Ly5wpTejPfDzp-8fI8/view?usp=sharing) or [here](https://huggingface.co/MonsterMMORPG/tools/tree/main)), unzip and place them in the ComfyUI/models/insightface/models/antelopev2 directory.

The main model can be downloaded from [HuggingFace](https://huggingface.co/InstantX/InstantID/resolve/main/ip-adapter.bin?download=true) and should be placed into the ComfyUI/models/instantid directory. (Note that the model is called ip_adapter as it is based on the [IPAdapter](https://github.com/tencent-ailab/IP-Adapter)).

You also needs a [controlnet](https://huggingface.co/InstantX/InstantID/resolve/main/ControlNetModel/diffusion_pytorch_model.safetensors?download=true), place it in the ComfyUI controlnet directory.

Remember at the moment this is only for SDXL.

I hope you can accept this request.

**Comment by Acly** [comment]
Because FaceID is already implemented, and InstantID provides too little (or any?) benefit.

**Reply by motolo** [reply]
Yes, the benefits are numerous, better management of styles simply with the prompt and not having to manage it with an another iPadapter for the style.  Furthermore, for the correct management of the position of the poses using zoe depth you obtain fantastic results.  I highly recommend watching the video by Matteo, author of ipdapter plus and InstantID... You too will notice the notable difference and what's more, the styles can be combined.  Writing simply, woman with green skin... the style of the prompt is copied and the face recreated is a huge plus:)
 If you can watch the video and you will notice the notable differences :)

https://www.youtube.com/watch?v=wMLiGhogOPE

Thanks for everything you do :)

**Comment by motolo** [comment]
![Schermata_20240802_210901](https://github.com/user-attachments/assets/21fb0e58-ee49-40a2-a8a6-5fb576ca8666)
